build_tools/benchmarks/common/benchmark_definition.py

```diff
@@ -441,45 +487,66 @@ class BenchmarkMetrics(object):
       framework.
   - cpu_time: the cpu time latency statistics returned by the benchmarking
       framework.
+  - host_memory: the host memory statistics returned by the benchmarking
+      framework.
+  - device_memory: the device memory statistics returned by the benchmarking
+      framework.
   - raw_data: additional JSON-compatible raw results returned by the
       benchmarking framework.
   """
   real_time: BenchmarkLatency
   cpu_time: BenchmarkLatency
+  host_memory: BenchmarkMemory
+  device_memory: BenchmarkMemory
   raw_data: Dict[str, Any]
 
   def to_json_object(self) -> Dict[str, Any]:
     return {
         "real_time": self.real_time.to_json_object(),
         "cpu_time": self.cpu_time.to_json_object(),
+        "host_memory": self.host_memory.to_json_object(),
+        "device_memory": self.device_memory.to_json_object(),
         "raw_data": self.raw_data,
     }
 
   @staticmethod
   def from_json_object(json_object: Dict[str, Any]):
     return BenchmarkMetrics(
-        BenchmarkLatency.from_json_object(json_object["real_time"]),
-        BenchmarkLatency.from_json_object(json_object["cpu_time"]),
-        json_object["raw_data"],
+        real_time=BenchmarkLatency.from_json_object(json_object["real_time"]),
+        cpu_time=BenchmarkLatency.from_json_object(json_object["cpu_time"]),
+        host_memory=BenchmarkMemory.from_json_object(
+            json_object["host_memory"]),
+        device_memory=BenchmarkMemory.from_json_object(
+            json_object["device_memory"]),
+        raw_data=json_object["raw_data"],
     )
 
 
-def parse_iree_benchmark_metrics(benchmark_stdout: str) -> BenchmarkMetrics:
+def parse_iree_benchmark_metrics(benchmark_stdout: str,
+                                 benchmark_stderr: str) -> BenchmarkMetrics:
   """Extract benchmark metrics from the output of iree-benchmark-module.
 
   Args:
     benchmark_stdout: The stdout of iree-benchmark-module with
       --benchmark_format=json.
+    benchmark_stdout: The stderr of iree-benchmark-module with
+      --print_statistics=true.
 
   Returns:
     A populated BenchmarkMetrics dataclass.
   """
-  iree_benchmark_json = json.loads(benchmark_stdout)
-  real_time, cpu_time = _get_google_benchmark_latencies(iree_benchmark_json)
+  benchmark_json = json.loads(benchmark_stdout)
+  real_time, cpu_time = _get_google_benchmark_latencies(benchmark_json)
   return BenchmarkMetrics(
       real_time=real_time,
       cpu_time=cpu_time,
-      raw_data=iree_benchmark_json,
+      host_memory=_get_iree_memory_statistics(benchmark_stderr, "HOST_LOCAL"),
+      device_memory=_get_iree_memory_statistics(benchmark_stderr,
+                                                "DEVICE_LOCAL"),
+      raw_data=dict(
+          google_benchmark_json=benchmark_json,
+          benchmark_stderr=benchmark_stderr,
```

Comment:
What is the size of the stderr? Will this significantly increase the size of JSON?

Reply:
~250 characters, so I'd say the increase isn't significant.

Reply:
```js
{
  'benchmark_stderr': '[[ iree_hal_allocator_t memory statistics ]]\n  HOST_LOCAL:            0B peak /            0B allocated /            0B freed /            0B live\nDEVICE_LOCAL:      7675392B peak /      7675392B allocated /            0B freed /      7675392B live',
}
```

Reply:
I mostly included it based off of your suggestion here: https://github.com/openxla/iree/pull/13206#discussion_r1175864645. I agree that it isn't particularly necessary in this case, as we're already extracting 100% of the information contained in the raw data. The only case where I could see this becoming useful would be if more memory tracking information was added down the line, but I don't think that's worth worrying about. WDYT?

Reply:
Yeah, since we include most of the information in our parsed data, I think the stderr doesn't need to be included.